src/ripple/rpc/handlers/GatewayBalances.cpp

```diff
@@ -0,0 +1,228 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/rpc/impl/AccountFromString.h>
+#include <ripple/rpc/impl/LookupLedger.h>
+#include <ripple/app/paths/RippleState.h>
+
+namespace ripple {
+
+// Query:
+// 1) Specify ledger to query.
+// 2) Specify issuer account (cold wallet) in "account" field.
+// 3) Specify accounts that hold gateway assets (such as hot wallets)
+//    using "hotwallet" field which should be either a string (if just
+//    one wallet) or an array of strings (if more than one).
+
+// Response:
+// 1) Array, "obligations", indicating the total obligations of the
+//    gateway in each currency. Obligations to specified hot wallets
+//    are not counted here.
+// 2) Object, "balances", indicating balances in each account
+//    that holds gateway assets. (Those specified in the "hotwallet"
+//    field.)
+// 3) Object of "assets" indicating accounts that owe the gateway.
+//    (Gateways typically do not hold positive balances. This is unusual.)
```

Comment:
Perhaps instead: `Gateways typically do not hold positive balances so this field is usually empty.`